astropy/io/ascii/cds.py

```diff
@@ -164,14 +210,437 @@ def get_cols(self, lines):
+                # minute/arcminute, second/arcsecond columns.
+                if 'ra' in col.representation_component_names.keys():
+                    ra_col, dec_col = col.ra.hms, col.dec.dms
+                    coords = [ra_col.h, ra_col.m, ra_col.s,
+                              dec_col.d, dec_col.m, dec_col.s]
+                    names = ['RAh', 'RAm', 'RAs', 'DEd', 'DEm', 'DEs']
+                    coord_units = [u.h, u.min, u.second,
+                                   u.deg, u.arcmin, u.arcsec]
+                    coord_descrip = ['Right Ascension (hour)', 'Right Ascension (minute)',
+                                     'Right Ascension (second)', 'Declination (degree)',
+                                     'Declination (arcmin)', 'Declination (arcsec)']
+                    for coord, name, coord_unit, descrip in zip(
+                        coords, names, coord_units, coord_descrip):
+                            # Have Sign of Declination only in the DEd column.
+                            if name in ['DEm', 'DEs']:
+                                coord_col = Column(list(np.abs(coord)), name=name,
+                                                   unit=coord_unit, description=descrip)
+                            else:
+                                coord_col = Column(list(coord), name=name, unit=coord_unit,
+                                                   description=descrip)
+                            self.cols.append(coord_col)
+
+                # For all other coordinate types, simply divide into two columns
+                # for latitude and longitude resp. with the unit used been as it is.
+                else:
+                    # Galactic coordinates.
+                    if col.name == 'galactic':
+                        lon_col = Column(col.l, name='GLON',
+                                         description='Galactic Longitude',
+                                         unit=col.representation_component_units['l'])
+                        lat_col = Column(col.b, name='GLAT',
+                                         description='Galactic Latitude',
+                                         unit=col.representation_component_units['b'])
+                    # Ecliptic coordinates, can be any of various available.
+                    else:
+                        lon_col = Column(col.lon, name='ELON',
+                                         description = 'Ecliptic Longitude (' + col.name + ')',
+                                         unit=col.representation_component_units['lon'])
+                        lat_col = Column(col.lat, name='ELAT',
+                                         description = 'Ecliptic Latitude (' + col.name + ')',
+                                         unit=col.representation_component_units['lat'])
+                    self.cols.append(lon_col)
+                    self.cols.append(lat_col)
+
+                self.cols.pop(i)  # Delete original ``SkyCoord`` column.
```

Comment:
We need to convert a SkyCoord mixin column into string columns in a very specific form for MRT/CDS tables. That conversion is above (details may still change). So, our options are: (1) Change the table to add new columns and  remove the SkyCoord. Leave it up to the user to pass in a copy of their original table if they want to keep the original or (2) we take care of making a copy in this code. That way, the users don't have to worry about it and their original table stays undisturbed.

I have a preference for (2) @taldcroft What do you think?

Reply:
My preference is (2) as well -- would like to hear @taldcroft's opinion!

Reply:
I have added a `table = table.copy()` in `Cds.write()` and a small test for this purpose.